liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/ServerFeatureSupport.java

```diff
@@ -191,6 +213,91 @@ private static boolean isSubModule(MavenProject potentialTopModule, MavenProject
+            	
+            	if (skipConfigured(reactorProject)) {
+            		getLog().debug("Skip configured on project: " + reactorProject.getArtifactId() + " - Ignoring");
+            	} else if (reactorProject.getPackaging().equals("jar")) {
+            		getLog().debug(reactorProject.getArtifactId() + " is a jar project - Ignoring");
+            	} else {
+                    leaves.add(reactorProject);
+            	}
+            }
+        }
+            
+        // At this point, the only leaves we should have is one final downstream project and the parent pom project. 
+        // Loop through and find any conflicts.
+        for (MavenProject leaf1 : leaves) {
+            for (MavenProject leaf2 : leaves) {
+            	// Check that the leaves are not the same module and that one of the leaves is not the parent.
+                if (leaf1 != leaf2 && !(isSubModule(leaf2, leaf1) || isSubModule(leaf1, leaf2)) ) {
+                    conflicts.add(leaf1);
+                    conflicts.add(leaf2);
+                }
+            }
+        }
+        
+        if (conflicts.isEmpty() ) {
+        	List<MavenProject> devModeProjects = new ArrayList<MavenProject>();
+        	for (MavenProject leaf : leaves) {
+        		devModeProjects.addAll(graph.getUpstreamProjects(leaf, true));
+        		devModeProjects.add(leaf);
+        	}
+        	
+        	getLog().debug("Resolved multi-module projects: ");
+        	for (MavenProject project : devModeProjects) {
+        		getLog().debug(project.getArtifactId());
+        	}
+        	return devModeProjects;
+        } else {
+        	
+        	List<String> conflictModuleRelativeDirs = new ArrayList<String>();
+            for (MavenProject conflict : conflicts) {
+                // Make the module path relative to the multi-module project directory
+                conflictModuleRelativeDirs.add(getModuleRelativePath(conflict));
+            }
+            
+            throw new MojoExecutionException("Found multiple independent modules in the Reactor build order: "
+                    + conflictModuleRelativeDirs
+                    + ". Skip a conflicting module in the Liberty configuration or specify the module containing the Liberty configuration that you want to use for the server by including the following parameters in the Maven command: -pl <module-with-liberty-config> -am");
+        }
+    }
+    
+    private boolean skipConfigured(MavenProject project) {
+    	
+        // Properties that are set in the pom file
+        Properties props = project.getProperties();
+
+        // Properties that are set by user via CLI parameters
+        Properties userProps = session.getUserProperties();
+    	Plugin libertyPlugin = getLibertyPluginForProject(project);
+        Xpp3Dom config = ExecuteMojoUtil.getPluginGoalConfig(libertyPlugin, "dev", getLog());
+        
+        return DevHelper.getBooleanFlag(config, userProps, props, "skip");
```

Comment:
A side effect of this new path is that, when running the new LMP code with liberty:dev with any LMP config at all, even with a single module project, I get an extra:

> [INFO] --- liberty:3.11-SNAPSHOT:dev (default-cli) @ intro ---
> [INFO] skip execution goal configuration validation for liberty-maven-plugin:dev

I think we should be less verbose (yes, harping on again with the INFO messages).

Actually looking at the logic in `io.openliberty.tools.maven.utils.ExecuteMojoUtil.validateConfiguration(Plugin, String, Xpp3Dom, Log)`  the logging almost looks backwards to me.

We warn people when we do NOT ignore the config they specified but we're quiet in the case in which we strip off only the elements we claim support for.   We're not really validating anything are we?  We're just filtering out config we support and ignoring stuff we don't.  @cherylking  what do you think here?

Reply:
Ok, I looked at the code printing that info log message. I would suggest two things:

1. Change the info log message to debug (and maybe reword it).
2. Do not print the log message at all for the `dev` goal. There's no purpose to that method for the `dev` goal. Or perhaps don't call that method for the `dev` goal? It's literally doing nothing.